### ISSUE TITLE
Error: ConnectFailed when attempting to communicate with a MB8600

### ISSUE THREAD

**Issue by arcasinky** [jdicioccio/moto_arris_cm_stats#1]
Hi.  I have an MB8600 (192.168.100.1).  Attempts to run moto_arris_cm_stats results in a ConnectFailed error.  I've verified that I can manually log into the modem's admin UI using a web browser.

Any ideas how to proceed?

**Comment by jdicioccio** [comment]
If GitHub notified me of this, then I completely missed it. Sorry! Did you manage to get this working? I know at one point, they switched my firmware to only use HTTPS. I'm wondering if perhaps yours might still be using HTTP.

**Comment by arcasinky** [comment]
Hi.  I never got your particular app working on my MB8600 but it gave me some hints about how to proceed.  Combined with some time spent with the Firefox debugger, I was eventually able to throw together a Python script that successfully logs in and extracts the stats.  My modem still uses HTTP so it's good to know that my ISP might push an update to only allow HTTPS.

One thing that's different between my code and yours is your appears to be able to access the HNAPs without logging-in.  I'm could've swore that failed when I tried something similar.

**Comment by jdicioccio** [comment]
Ah interesting. I guess each ISP is going to have variations in their firmware. As for the HNAPs: I initially was passing headers for auth, but then noticed that it didn't seem to matter in my case. So, I started trimming away headers until it stopped working. It turns out that with Comcast, you seem to barely need any.

In any case, as of recently, I'm no longer using that modem at all, so I suspect that that code path may rot unless I've got some secret users out there.
